models/m_audit.erl

```diff
@@ -131,6 +135,44 @@ user_agent_id(UserAgent, Context) ->
     Transaction = fun() -> z_db:transaction(F, Context) end,
     z_depcache:memo(Transaction, {user_agent_id, BUA}, ?MAXAGE_UA_STRING, Context).
 
+% Remove entries older than 2 years.
+periodic_cleanup(Context) ->
+    z_db:q("delete from audit where id in (select id from audit where created < now() - interval '2 years' limit 10000)", Context, 300000).
+
+
+% Update all records in the audit table
+update_audit_table(Context) ->
+    io:fwrite(standard_error, "Updating audit table", []),
+    update_audit_table1(Context).
+
+update_audit_table1(Context) ->
+    case props_json_update(Context) of
+        {ok, N} when N > 0 ->
+            io:fwrite(standard_error, ".", []),
+            update_audit_table1(Context);
+        {ok, 0} ->
+            io:fwrite(standard_error, ".~n", []),
+            done
+    end.
```

Comment:
Why the extra function? Seems like L146:148 could be deleted to remove the indirection.

Why not use `zDebug()` instead of `io:fwrite(standard_error, …)`? I understand that it is doing a kind of progress bar, but that is going to standard error. Seems it would be better to provide `zDebug` lines for diagnosis, and a final `zInfo`/`zError` on completion.

Reply:
It is a function which you only need to run once to get the data transferred. I did not want to spend the time to get a web-interface for this. Therefor I made it especially for the shell only. No debug statements which go to a file. The dots are only for operator feedback. There is an extra level of indirection because of that too. You only need to print "Updating audit table" once.